packages/plugins/embarkjs/src/index.js

```diff
@@ -46,11 +57,62 @@ class EmbarkJS {
     this.events.setCommandHandler("embarkjs:contract:generate", this.addContractArtifact.bind(this));
     this.events.setCommandHandler("embarkjs:contract:runInVm", this.runInVm.bind(this));
 
+    if (!this.enabled) return;
+
     embark.registerActionForEvent("pipeline:generateAll:before", this.addEmbarkJSArtifact.bind(this));
     embark.registerActionForEvent("pipeline:generateAll:before", this.addContractIndexArtifact.bind(this));
+
+    if (this.config.blockchainConfig.enabled) {
+      this.setupEmbarkJS();
+    }
+
+    embark.registerActionForEvent("deployment:contract:deployed", {priority: 40}, this.registerInVm.bind(this));
+    embark.registerActionForEvent("deployment:contract:undeployed", this.registerInVm.bind(this));
+    embark.registerActionForEvent("deployment:contract:deployed", this.registerArtifact.bind(this));
+    embark.registerActionForEvent("deployment:contract:undeployed", this.registerArtifact.bind(this));
+  }
+
+
+
+  async setupEmbarkJS() {
+    if (!this.enabled) return;
```

Comment:
This `if` might not be needed either, since we only call `setupEmbarkJS` if we are enabled.